plugins/action/common/prepare_plugins/prep_002_global.py

```diff
@@ -0,0 +1,115 @@
+            # created and populated with the data from the old global key if it exists.
+            # If the new global key already exists and has data, it will not be overwritten.
+            if new_global_key in dm_check['keys_not_found'] or new_global_key in dm_check['keys_no_data']:
+                deprecated_msg = (
+                    f"Attempting to use vxlan.global keys due to vxlan.global.{new_global_key} not being defined. "
+                    f"General vxlan.global keys are being deprecated. Please use vxlan.global.{new_global_key}."
+                )
+                display.deprecated(msg=deprecated_msg, version="1.0.0", collection_name='cisco.nac_dc_vxlan')
+
+                model_data['vxlan']['global'].update({new_global_key: {}})
+
+                for key in BACKWARD_COMPATIBLE_KEYS:
+                    if key in model_data['vxlan']['global']:
+                        model_data['vxlan']['global'][new_global_key].update({key: model_data['vxlan']['global'][key]})
+                        model_data['vxlan']['global'].pop(key, None)
+
+                return self.kwargs['results']
+
+            # This elif handles the case where the new global key exists but is empty or has data while data still exists
+            # under the old global key. This is to ensure that the new global key is populated with the data from the old
+            # global key if it exists and only if the new global key does not already have data at the key location.
+            elif new_global_key in dm_check['keys_found'] and (new_global_key in dm_check['keys_data'] or new_global_key in dm_check['keys_no_data']):
+                for key in BACKWARD_COMPATIBLE_KEYS:
+                    # Check if the key exists in the new global key
+                    dm_check = data_model_key_check(model_data, PARENT_KEYS + [key])
+                    # If the key does not exist or has no data in the new global key, we can safely try to copy the data
+                    # from the old global key to the new global key if the old global key has data
+                    if key in dm_check['keys_not_found'] or key in dm_check['keys_no_data']:
+                        # Check if the key exists in the old global key
+                        dm_check = data_model_key_check(model_data, PARENT_KEYS[:-1] + [key])
+                        # If the key exists and has data in the old global key, we can copy the data
+                        if key in dm_check['keys_found'] and key in dm_check['keys_data']:
+                            model_data['vxlan']['global'][new_global_key].update({key: model_data['vxlan']['global'][key]})
+                            model_data['vxlan']['global'].pop(key, None)
+                    elif key in dm_check['keys_found'] and key in dm_check['keys_data']:
+                        model_data['vxlan']['global'].pop(key, None)
+
+                return self.kwargs['results']
+
+        return self.kwargs['results']
```

Comment:
Not merge blocking but maybe consider a rule if values are defined in both the old and new locations

Reply:
Not merge blocking